src/support.h

```diff
@@ -30,6 +30,40 @@ class AreaSupport {
      * \param layer_count total number of layers
      */
     static void generateSupportAreas(SliceDataStorage& storage, unsigned int layer_count);
+
+    /*!
+     * Generate gradual support on the already generated support areas. This must be called after generateSupportAreas().
+     * This uses the same technic as the gradual infill.
+     * \param storage data storage containing the input layer outline data and containing the output support storage per layer
+     * \param total_layer_count total number of layers
+     * \param gradual_support_step_height The height difference between consecutive density support areas
+     * \param max_support_steps the maximum exponent of division of support density. At 5 the least dense support will be 2^4 * infill_line_distance i.e. one 16th as dense
+     */
+    static void generateGradualSupport(SliceDataStorage& storage, unsigned int total_layer_count, unsigned int gradual_support_step_height, unsigned int max_support_steps);
+
+    /*!
+     * \brief Combines the support infill of multiple layers.
+     * 
+     * The support infill layers are combined while the thickness of each layer is
+     * multiplied such that the infill should fill up again to the full height of
+     * all combined layers.
+     * 
+     * \param storage data storage containing the input layer outline data and containing the output support storage per layer
+     * \param total_layer_count The total number of layers.
+     * \param combine_layers_amount The number of layers to combine.
+     */
+    static void combineSupportInfillLayers(SliceDataStorage& storage, unsigned int total_layer_count, unsigned int combine_layers_amount);
+
+    /*!
+     * Generate the insets of the given support infill outline.
+     *
+     * \param insets The insets result to output.
```

Comment:
You should specify that it is an output parameter like so:
`\param[out] insets [description]`

Parameters which are both input and output parameters are tagged as such:
`\param[in,out] ...` without a space after the comma. (otherwise doxygen won't recognize it)